sonar-core/src/main/resources/org/sonar/core/measure/db/MeasureMapper.xml

```diff
@@ -83,4 +83,11 @@
     )
   </insert>
 
+  <select id="selectMetricKeysForSnapshot" parameterType="long" resultType="string">
+  SELECT m.name
```

Comment:
A snapshot can have multiple measures with the same metric id (key is [snapshot_id, metric_id, characteristic_id, rule_id, person_id]). For this reason SELECT DISTINCT is probably missing here.